qiita_db/support_files/patches/38.sql

```diff
@@ -0,0 +1,241 @@
+		   ('waiting', 'The job is waiting for a previous job in the workflow to be completed in order to be executed.');
+
+-- Populate the newly created tables
+DO $do$
+DECLARE
+	in_slq_param_id		bigint;
+	in_sl_param_id		bigint;
+	in_po_param_id		bigint;
+	dflt_slq_id			bigint;
+	dflt_sl_id			bigint;
+	dflt_per_sample_id	bigint;
+	dflt_po_id			bigint;
+BEGIN
+	-- Add the artifact type information for the input parameters for the commands
+	-- command_id = 1 -> Split libraries FASTQ
+	SELECT command_parameter_id FROM qiita.command_parameter
+		WHERE command_id = 1 AND parameter_name = 'input_data'
+		INTO in_slq_param_id;
+
+	-- Split libraries FASTQ supports FASTQ (3) and per_sample_FASTQ (5)
+	INSERT INTO qiita.parameter_artifact_type (command_parameter_id, artifact_type_id)
+		VALUES (in_slq_param_id, 3), (in_slq_param_id, 5);
+
+	-- command_id = 2 -> Split libraries
+	SELECT command_parameter_id FROM qiita.command_parameter
+		WHERE command_id = 2 AND parameter_name = 'input_data'
+		INTO in_sl_param_id;
+
+	-- Split libraries supports SFF (1), FASTA_Sanger (2), FASTA (4)
+	INSERT INTO qiita.parameter_artifact_type (command_parameter_id, artifact_type_id)
+		VALUES (in_sl_param_id, 1), (in_sl_param_id, 2), (in_sl_param_id, 4);
+
+	-- command_id = 3 -> Pick closed-reference OTUs
+	SELECT command_parameter_id FROM qiita.command_parameter
+		WHERE command_id = 3 AND parameter_name = 'input_data'
+		INTO in_po_param_id;
+
+	-- Pick closed-reference OTUs supports Demultiplexed (6)
+	INSERT INTO qiita.parameter_artifact_type (command_parameter_id, artifact_type_id)
+		VALUES (in_po_param_id, 6);
+
+
+	-- Add the output information for each command
+	INSERT INTO qiita.command_output (name, command_id, artifact_type_id)
+		VALUES ('demultiplexed', 1, 6), ('demultiplexed', 2, 6), ('OTU table', 3, 7);
+
+
+	-- Add the default workflow for the target gene pipeline.
+	-- We are going to create three different default workflows for the
+	-- target gene plugin:
+	--   1) FASTQ upstream workflow: split_libraries_fastq.py + OTU picking
+	--   2) FASTA upstream workflow: split_libraries.py + OTU picking
+	--   3) Per sample FASTQ upstream workflow:
+	--      split_libraries_fastq.py + OTU picking using per sample fastq parameters
+	-- In order to choose the default parameters set, we are going to choose
+	-- the one with minimum id. The reason for this is that in the live system
+	-- there are default parameter set that were added manually, so we don't
+	-- know the ids for those parameter set. Note that we do know
+	-- the command id because they're inserted in patch 33.sql, and that is
+	-- the only way of adding commands at this point.
+
+	-- Insert default workflow
+	INSERT INTO qiita.default_workflow (software_id, name)
+		VALUES (1, 'FASTQ upstream workflow'),
+			   (1, 'FASTA upstream workflow'),
+			   (1, 'Per sample FASTQ upstream workflow');
+
+	-- Retrieve all the ids of the default parameter set that we need
```

Comment:
Those are "all" that we need here. Note that each query returns a set but I'm getting the minimum of those, specifically because at the patch level there is no control over these.